src/lib/falcor/FalcorController.js

```diff
@@ -34,41 +34,76 @@ export default class FalcorController extends BaseComponent {
   loadFalcorCache() {
     console.log("LOADING FROM CACHE")
-    const falcorPath = this.constructor.getFalcorPath(this.props.params)
-
-    const data = this.props.model.getCache(falcorPath)
+    let falcorPath = this.constructor.getFalcorPath(this.props.params)
+    // If the component doesn't want any data
+    if (!falcorPath || falcorPath.length === 0) {
+      console.log("NO PATHS WERE GIVEN");
+      this.safeSetState({
+        ready: true,
+        data: []
+      })
+      return;
+    }
+    // If we're only passing a single pathSet we compensate for the spread operator
+    if (!(falcorPath[0] instanceof Array)) {
+      falcorPath = [falcorPath];
+    }
+    
+    const data = expandCache(this.props.model.getCache(...falcorPath));
     if (data) {
       this.safeSetState({
         ready: true,
         data: data
       })
     } else {
-      throw new Error("Serverside render of component: " + this.constructor.name +
-                      " failed. Data not in cache")
+      console.error("Serverside render of component: " + this.constructor.name +
+                      " failed. Data not in cache. Error is on components state");
+      this.safeSetState({
+        ready: true,
+        data: null,
+        error: new Error("Serverside render of component: " + this.constructor.name +
+                      " failed. Data not in cache. Falcor Path attempted fetched was: " + JSON.stringify(falcorPath))
+      });
     }
   }
 
   // Makes falcor fetch its paths
   falcorFetch(falcorPath) {
     console.log("STARTING FETCH")
-
+    // If the component doesn't want any data
+    if (!falcorPath || falcorPath.length === 0) {
+      console.log("NO PATHS WERE GIVEN");
+      this.safeSetState({
+        ready: true,
+        data: null
+      })
+      return;
+    }
+    // If we're only passing a single pathSet we compensate for the spread operator
+    if (!(falcorPath[0] instanceof Array)) {
+      falcorPath = [falcorPath];
+    }
     // Then if we are a client, we can try to do a fetch as well
     this.safeSetState({fetching: false})
-    this.props.model.get(falcorPath).then((x) => {
+    this.props.model.get(...falcorPath).then((x) => {
       console.log("FALCOR FETCH COMPLETED")
       if (x) {
         this.safeSetState({
           ready: true,
           fetching: false,
           data: x.json
-        })
-      } else {
-        throw new Error("FalcorPath: " + falcorPath + " returned no data")
+        });
       }
-    }).catch((e) => {
-      console.error("Failed to fetch for falcorPath: " + falcorPath)
-      console.error(e.stack)
-    })
+      else {
+        console.error("FalcorPath: " + JSON.stringify(falcorPath) + " returned no data. The error is on the components state.");
+        this.safeSetState({
+          ready: true,
+          fetching: false,
+          data: null,
+          error: new Error("FalcorPath: " + JSON.stringify(falcorPath) + " returned no data")
+        });
+      }
+    });
```

Comment:
I can't comment there because of github, but on line 116, deete this.safeSetState, we should move this into the falcorFetchFunction itself.